Doc/reference/expressions.rst

```diff
@@ -83,18 +83,131 @@ exception.
    pair: name; mangling
    pair: private; names
 
-**Private name mangling:** When an identifier that textually occurs in a class
-definition begins with two or more underscore characters and does not end in two
-or more underscores, it is considered a :dfn:`private name` of that class.
-Private names are transformed to a longer form before code is generated for
-them.  The transformation inserts the class name, with leading underscores
-removed and a single underscore inserted, in front of the name.  For example,
-the identifier ``__spam`` occurring in a class named ``Ham`` will be transformed
-to ``_Ham__spam``.  This transformation is independent of the syntactical
-context in which the identifier is used.  If the transformed name is extremely
-long (longer than 255 characters), implementation defined truncation may happen.
-If the class name consists only of underscores, no transformation is done.
+Private name mangling
+^^^^^^^^^^^^^^^^^^^^^
 
+When an identifier that textually occurs in a class definition begins with two
```

Comment:
I think the detailed discussion of imports below should be replaced with a more precise definition of what an "identifier" is here. It is not obvious that it includes attribute names and imported module names, as well as directly used names.

Reply:
From looking at the implementation, the names that are mangled are:
* Any name used as the name of a variable that is assigned or read. However, the `__name__` attribute of nested functions, classes, and type aliases is not mangled.
* Any attribute accessed
* The name of imported modules and of names that are imported from a module, unless the module is part of a package (i.e., the name contains a dot).